### PR TITLE
73X: L1REPACK for HLT and fixes

### PR DESCRIPTION
**Pull request by Martin-Grunewald**
73X: L1REPACK for HLT and fixes
Backport of #6979 including also #6847 except for the HI menu itself which is in #6960

### CODE REVIEW

**Comment by Martin-Grunewald** [comment]
@ktf @nclopezo
Also this PR is stuck in "testing" nearly two weeks - can you please check?

**Comment by ktf** [comment]
please test

**Comment by civanch** [comment]
+1

**Comment by Martin-Grunewald** [comment]
Can we please move on this PR?

**Comment by ktf** [comment]
+1

**Comment by ktf** [comment]
@mulhearn @davidlange6 waiting for you...

**Comment by davidlange6** [comment]
i was trying to convince myself that nothing in prod gets changed by this PR, but I didn't manage yet… will try again later.

**Comment by davidlange6** [comment]
Hi Martin, Mike, all

I'm down to one line:

--- a/L1Trigger/L1TCommon/python/customsPostLS1.py
+++ b/L1Trigger/L1TCommon/python/customsPostLS1.py
@@ -30,6 +30,11 @@ def customiseSimL1EmulatorForPostLS1(process):
         process.gctDigiToRaw.gctInputLabel = 'simCaloStage1LegacyFormatDigis'
 
     if hasattr(process, 'simGctDigis'):
+        for sequence in process.sequences:
+            #print "INFO:  checking sequence ", sequence
+            #print "BEFORE:  ", getattr(process,sequence)
+            getattr(process,sequence).replace(process.simGctDigis,process.L1TCaloStage1)
+            #print "AFTER:  ", getattr(process,sequence)

could you summarize what this is doing?

Cheers-
David

**Comment by mulhearn** [comment]
Sorry, was AWOL:  2 jet lagged babies, start of new quarter, and no
internet at home...

This is replacing the legacy GCT step with the post LS1 step in all
sequences.

-Mike

**Comment by Martin-Grunewald** [comment]
Hi,

This used to be run only for paths, which means sequences in paths
containing the module were expanded (modules put directly into the
path and the intermediate sequence definition removed). This way the 
sequences themselves are kept and can thus be used in further 
manupulations.

Best regards

Martin

On Sat, 10 Jan 2015, mulhearn wrote:

> Sorry, was AWOL:  2 jet lagged babies, start of new quarter, and no
> internet at home...
> 
> This is replacing the legacy GCT step with the post LS1 step in all
> sequences.
> 
> -Mike
> 
> On Sat, Jan 10, 2015 at 2:59 AM, David Lange <notifications@github.com>
> wrote:
> 
> > Hi Martin, Mike, all
> >
> > I'm down to one line:
> >
> > --- a/L1Trigger/L1TCommon/python/customsPostLS1.py
> > +++ b/L1Trigger/L1TCommon/python/customsPostLS1.py
> > @@ -30,6 +30,11 @@ def customiseSimL1EmulatorForPostLS1(process):
> > process.gctDigiToRaw.gctInputLabel = 'simCaloStage1LegacyFormatDigis'
> >
> > if hasattr(process, 'simGctDigis'):
> > + for sequence in process.sequences:
> > + #print "INFO: checking sequence ", sequence
> > + #print "BEFORE: ", getattr(process,sequence)
> > +
> > getattr(process,sequence).replace(process.simGctDigis,process.L1TCaloStage1)
> >
> > + #print "AFTER: ", getattr(process,sequence)
> >
> > could you summarize what this is doing?
> >
> > Cheers-
> > David
> >
> >
> > On Jan 9, 2015, at 4:07 PM, David Lange <David.Lange@cern.ch> wrote:
> >
> > > i was trying to convince myself that nothing in prod gets changed by
> > this PR, but I didn't manage yet… will try again later.
> > >
> > > On Jan 9, 2015, at 4:05 PM, Giulio Eulisse <notifications@github.com>
> > > wrote:
> > >
> > >> @mulhearn @davidlange6 waiting for you...
> > >>
> > >> —
> > >> Reply to this email directly or view it on GitHub.
> > >>
> > >
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <https://github.com/cms-sw/cmssw/pull/6980#issuecomment-69451967>.
> >
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/cms-sw/cmssw/pull/6980#issuecomment-69463617


Martin

--

Martin W. Gruenewald           e-mail: Martin.Grunewald@cern.ch
                        http://cern.ch/Martin.Grunewald

**Comment by davidlange6** [comment]
@Martin-Grunewald - I see now. Thanks - I can confirm we don't get any (real) changes to the config
